controllers/monocle_controller.go

```diff
@@ -58,6 +62,80 @@ func (r *MonocleReconciler) rollOutWhenApiSecretsChange(ctx context.Context, log
 	return nil
 }
 
+func (r *MonocleReconciler) triggerJobWhenConfigMapChange(ctx context.Context, logger logr.Logger, depl appsv1.Deployment, apiConfigVersion string) error {
+	// Check if Annotation has changed
+	previousVersion := depl.Annotations["apiConfigVersion"]
+	if previousVersion != apiConfigVersion {
+
+		logger.Info("Start a job to configmap update",
+			"name", depl.Name,
+			"previous configmap version", previousVersion,
+			"new configmap version", apiConfigVersion)
+		depl.Annotations["apiConfigVersion"] = apiConfigVersion
+		// Update Deployment Resource
+		err := r.Update(ctx, &depl)
+		if err != nil {
+			return err
+		}
+
+		// Get Api container from the Deployment Resource
```

Comment:
Why not just having the api container definition stored in a var "apiContainer" on top level and just reuse that var for api deployment and job definitions ? This will also prevent the needs for regexp lib import.